src/Compilers/CSharp/Test/Semantic/Semantics/RefExtensionMethodsTests.cs

```diff
@@ -0,0 +1,1844 @@
+  {
+    .custom instance void [mscorlib]System.Runtime.CompilerServices.ExtensionAttribute::.ctor() = ( 01 00 00 00 )
+    .param [1]
+    .custom instance void System.Runtime.CompilerServices.IsReadOnlyAttribute::.ctor() = ( 01 00 00 00 )
+    .maxstack  8
+    IL_0000:  nop
+    IL_0001:  ldarg.0
+    IL_0002:  ldind.i4
+    IL_0003:  call       void [mscorlib]System.Console::Write(object)
+    IL_0008:  nop
+    IL_0009:  ret
+  }
+}");
+
+            var code = @"
+public static class Program
+{
+    public static void Main()
+    {
+        string x = ""test"";
+        x.PrintValue();
+        Extensions.PrintValue(x);
+    }
+}";
+
+            CreateCompilationWithMscorlibAndSystemCore(code, references: new[] { reference }).VerifyDiagnostics(
+                // (7,11): error CS1061: 'string' does not contain a definition for 'PrintValue' and no extension method 'PrintValue' accepting a first argument of type 'string' could be found (are you missing a using directive or an assembly reference?)
+                //         x.PrintValue();
+                Diagnostic(ErrorCode.ERR_NoSuchMemberOrExtension, "PrintValue").WithArguments("string", "PrintValue").WithLocation(7, 11));
+        }
+
+        [Fact]
+        public void RefReadOnlyExtensionMethodsReceiverTypes_InterfaceConstrainedGenericTypes_NotAllowed_IL()
+        {
+            var reference = CompileIL(ExtraRefReadOnlyIL + @"
+.class public abstract auto ansi sealed beforefieldinit Extensions extends [mscorlib]System.Object
+{
+  .custom instance void [mscorlib]System.Runtime.CompilerServices.ExtensionAttribute::.ctor() = ( 01 00 00 00 )
+  .method public hidebysig static void  PrintValue<([mscorlib]System.IComparable) T>(!!T& p) cil managed
+  {
+    .custom instance void [mscorlib]System.Runtime.CompilerServices.ExtensionAttribute::.ctor() = ( 01 00 00 00 )
+    .param [1]
+    .custom instance void System.Runtime.CompilerServices.IsReadOnlyAttribute::.ctor() = ( 01 00 00 00 )
+    .maxstack  8
+    IL_0000:  nop
+    IL_0001:  ldarg.0
+    IL_0002:  ldind.i4
+    IL_0003:  call       void [mscorlib]System.Console::Write(object)
+    IL_0008:  nop
+    IL_0009:  ret
+  }
+}");
+
+            var code = @"
+public static class Program
+{
+    public static void Main()
+    {
+        string x = ""test"";
+        x.PrintValue();
+        Extensions.PrintValue(x);
+    }
+}";
+
+            CreateCompilationWithMscorlibAndSystemCore(code, references: new[] { reference }).VerifyDiagnostics(
+                // (7,11): error CS1061: 'string' does not contain a definition for 'PrintValue' and no extension method 'PrintValue' accepting a first argument of type 'string' could be found (are you missing a using directive or an assembly reference?)
+                //         x.PrintValue();
+                Diagnostic(ErrorCode.ERR_NoSuchMemberOrExtension, "PrintValue").WithArguments("string", "PrintValue").WithLocation(7, 11));
```

Comment:
>Diagnostic(ErrorCode.ERR_NoSuchMemberOrExtension, "PrintValue").WithArguments("string", "PrintValue").WithLocation(7, 11)); [](start = 16, length = 123)

We should test a scenario like that when the PrintValue is declared in a referenced compilation rather than in IL. I would expect to get similar error.